### PR TITLE
feat(eslint-config-carbon): add storybook config

### PR DESCRIPTION
**Pull request by joshblack**
This PR introduces an ESLint configuration for `eslint-plugin-storybook`. This plugin enables a variety of rules for the style and/or convention of our storybook. These rules will make sure our stories are structured consistently for Component Story Format v3.

#### Changelog

**New**

- Add `eslint-plugin-storybook` to our ESLint configuration and corresponding config

**Changed**

- Updated files based on new rules, more often than not this fell into two categories:
  - Moving `parameters.component` to `component`
  - Updating stories to be PascalCase

**Removed**

#### Testing / Reviewing

- Verify the plugin is running on story files (`*.stories.js`, `*-story.js`)

### CODE REVIEW

File: packages/carbon-react/src/components/Icons/Icons.stories.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import './Icons.stories.scss';
 import React from 'react';
 import { Bee, Bicycle, ChevronUp } from '../../../icons';
 
+// eslint-disable-next-line storybook/csf-component
```
Comment (abbeyhrt) [inline packages/carbon-react/src/components/Icons/Icons.stories.js:12]: Why do these lines need to be disabled?

**Review by abbeyhrt — APPROVED** [review]
Looks good to me!

File: packages/carbon-react/src/components/Icons/Icons.stories.js (hunk 1 above)
Comment (joshblack) [reply packages/carbon-react/src/components/Icons/Icons.stories.js:12]: @abbeyhrt basically the rule is that every story _must_ have a `component:` key, which is great except we sometimes have stories that aren't for a specific component. This rule therefore doesn't really apply and so we ignore it 👀

**Comment by dakahn** [comment]
looks like we've got a storybook error about missing stories

**Comment by joshblack** [comment]
@dakahn should be good to go now!

**Review by dakahn — APPROVED** [review]

**Pull request merged**